.github/scripts/trymerge.py

```diff
@@ -1806,6 +1808,12 @@ def post_comment(msg: str) -> None:
         gh_post_commit_comment(pr.org, pr.project, commit_sha, revert_msg)
         gh_update_pr_state(pr.org, pr.project, pr.pr_num)
 
+        reviewers = pr.get_approved_by()
+        for author in pr.get_authors().keys():
+            if not gh_check_write_access(pr.org, pr.project, author):
```

Comment:
nit: You might be able to update the gql for GH_COMMIT_AUTHORS_FRAGMENT to return the author's write access as well, removing the need for this api call to github too

For context, in the past we've had issues with being throttled by github for making too many api calls, so it's a good practice to be mindful about when we have to make them

In this case, it's okay to not implement the above since this is only invoked during reverts (which is a relatively rare occurrence), but we'd want to be careful about adding logic like this to the normal merge operation